depends/Makefile

```diff
@@ -119,20 +119,24 @@ build_id_string:=$(BUILD_ID_SALT)
 # GCC only prints COLLECT_LTO_WRAPPER when invoked with just "-v", but we want
 # the information from "-v -E -" as well, so just include both.
 #
-# '3>&1 1>&2 2>&3 > /dev/null' is supposed to swap stdin and stdout and silence
-# stdin, since we only want the stderr output
-build_id_string+=$(shell $(build_CC) -v < /dev/null 3>&1 1>&2 2>&3 > /dev/null) $(shell $(build_CC) -v -E - < /dev/null 3>&1 1>&2 2>&3 > /dev/null)
-build_id_string+=$(shell $(build_AR) --version 2>/dev/null) $(filter AR_%,$(full_env)) ZERO_AR_DATE=$(ZERO_AR_DATE)
-build_id_string+=$(shell $(build_CXX) -v < /dev/null 3>&1 1>&2 2>&3 > /dev/null) $(shell $(build_CXX) -v -E - < /dev/null 3>&1 1>&2 2>&3 > /dev/null)
-build_id_string+=$(shell $(build_RANLIB) --version 2>/dev/null) $(filter RANLIB_%,$(full_env))
-build_id_string+=$(shell $(build_STRIP) --version 2>/dev/null) $(filter STRIP_%,$(full_env))
+# When invoking a shell, GNU Make special-cases exit code 127 (command not
+# found) by not capturing the output but instead passing it through. This is not
+# done for any other exit code.
+#
+# Therefore, we require a "|| true" to avoid this behaviour when in an
+# environment where the build_* or host_* may not exist yet.
+build_id_string+=$(shell $(build_CC) -v 2>&1 || true) $(shell $(build_CC) -v -E -o /dev/null - < /dev/null 2>&1 || true)
```

Comment:
Why adding `-o /dev/null` (here and other places)?

From GCC [docs](https://gcc.gnu.org/onlinedocs/gcc-9.3.0/gcc/Overall-Options.html#Overall-Options):
> The output is in the form of preprocessed source code, which is sent to the standard output.
> Input files that don’t require preprocessing are ignored.

That means no output for `-E` when passing `/dev/null` as an input, right?

Reply:
When I execute: `gcc -E -v -xc++ - < /dev/null | xclip -sel clip`, here's what I get in my clipboard:
```
# 1 "<stdin>"
# 1 "<built-in>"
# 1 "<command-line>"
# 1 "/usr/include/stdc-predef.h" 1 3 4
# 1 "<command-line>" 2
# 1 "<stdin>"
```